psi4/driver/driver_nbody.py

```diff
@@ -225,65 +211,188 @@ def nbody_gufunc(func, method_string, **kwargs):
     #    psioh = core.IOManager.shared_object()
     #    psioh.set_specific_retention(97, True)
 
-
-    bsse_str = bsse_type_list[0]
-    if len(bsse_type_list) >1:
-        bsse_str =  str(bsse_type_list)
+    bsse_str = metadata['bsse_type_list'][0]
+    if len(metadata['bsse_type_list']) > 1:
+        bsse_str =  str(metadata['bsse_type_list'])
     core.print_out("\n\n")
     core.print_out("   ===> N-Body Interaction Abacus <===\n")
     core.print_out("        BSSE Treatment:                     %s\n" % bsse_str)
 
+    # Build size and slices dictionaries
+    metadata['fragment_size_dict'] = {frag: metadata['molecule'].extract_subsets(frag).natom() for
```

Comment:
It doesn't look like this is used in `build_nbody_compute_list` so its a good candidate to be constructed inside `component_results` so that it has a signature of:
```
component_results = compute_nbody_components(func, method_string, mol, metadata)
```

Reply:
It could potentially be. I mostly just signaled this out as you are specifically inserting extra requirements into `metadata` not covered by a function which makes it more difficult to use. I think when you compute QM results a molecule should be associated with the data no matter what you do so when you sum up the results you know exactly what molecule you are dealing with. Happy to discuss this more, but I have no real thoughts on which path to take to solve this- up to you.

Reply:
Makes sense.  I'm all in favor of removing complication in the dict contents expected to be passed -- but I agree, passing the molecule around makes it so much more clear exactly what is going on.  This might actually be worth thinking about in the context of passing around a QC JSON object instead of custom `dict`'s, too.